models/index/writer.go

```diff
@@ -27,4 +27,6 @@ type Writer interface {
 	Events(height uint64, events []flow.Event) error
 	Payloads(height uint64, paths []ledger.Path, values []*ledger.Payload) error
 	Height(blockID flow.Identifier, height uint64) error
+	Transactions(blockID flow.Identifier, transactions []flow.Transaction) error
+	Collections(blockID flow.Identifier, collections []flow.LightCollection) error
```

Comment:
So the first transaction maps block ID to transaction IDs, and then each transaction ID to its transaction?

The second one maps blockID to collection IDs and then each collection ID to transaction IDs?

As mentioned earlier, I wouldn't really consider collections as an entity; it's just a double additional index. So maybe we can just do all of this in a single function, because these are just different mappings for transactions.